spec-draft.md

```diff
@@ -272,10 +272,65 @@ as follows:
 * The virtual keyboard: use sendKeys
 * acceleromator: TODO @mdas is working on this
 * geolocation: use regular webdriver endpoints
-* battery level: not in spec, perhaps exposed via executeScript
 * network speed: not in spec, perhaps exposed via executeScript
 
 
+Battery
+-------
+Device battery changes might trigger events on users applications, controlling this will help monitor and verify how apps responds to these factors.
+
+The battery state should have two options: ON, OFF depending if the power ac state it's charging or not.
+
+* GET /session/:sessionId/appium/device/battery_state
+  * returns BatteryState
+
+
+* POST /session/:sessionId/appium/device/battery_state
+  * accepts BatteryState
+
+
+A `BatteryState` can be one of two strings:
+ * ON
+ * OFF
+
+In some cases, apps might respond different on low battery level so the idea is to provide a method to set specific battery level for the devices.
+
+* GET /session/:sessionId/appium/device/battery_level
+  * returns [0..100]
+
+* POST /session/:sessionId/appium/device/battery_level
+  * accepts [0..100]
+
+
+External communication factors
+------------------------------
+Receiving messages or phone calls it's usual on mobile devices, having the possibility to mock this events would improve the application reaction on this external factors.
+
+### Phone calls(GSM)
+The idea is to be able to mock phone calls entering the device and handle the actions that triggers; you may want to accept it, cancel or put it on hold.
+
+* POST /session/:sessionId/appium/device/gsm_call
+  * accepts PhoneNumber, GsmAction
+
+
+`GsmAction`'s would be:
+
+* CALL
+* ACCEPT
+* CANCEL
+* HOLD
+
+The `CALL` action should trigger the mocked call to the device from the `PhoneNumber` provided, then another action should take place in order to complete the flow as in real life.
+
+
+### SMS
+During this days a lot of apps depend on receiving sms, even if they are not messaging platform, apps may want to check sms to validate accounts, send promotion codes, etc. We should add
+
+* POST /session/:sessionId/appium/device/sms_send
```

Comment:
I think just `POST .../sms` is fine; POST has the semantics of "creating" an sms on the device. but this is a nit.